internal/api/server/http_socket.go

```diff
@@ -33,6 +26,12 @@ func runAPIServer(routes *[]core.Route) (err error) {
 		err = errors.Join(err, fmt.Errorf("API server panicked: '%v'.", value))
 	}()
 
+	common.Lock(API_SERVER_LOCK)
```

Comment:
Nit (I would normally not hold up an approval for this, but there is another change you need to make so I may as well include it).

Add a comment here saying that we don't defer the unlock, but we unlock on all possible code paths.